packages/rest/src/body-parsers/body-parser.ts

```diff
@@ -0,0 +1,176 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as debugModule from 'debug';
+import {Request, RequestBodyParserOptions} from '../types';
+import {
+  inject,
+  Context,
+  instantiateClass,
+  Constructor,
+} from '@loopback/context';
+import {isReferenceObject, OperationObject} from '../..';
+import {is} from 'type-is';
+import {RestHttpErrors} from '../rest-http-error';
+
+const debug = debugModule('loopback:rest:body-parser');
+
+import {
+  RequestBody,
+  BodyParser,
+  REQUEST_BODY_PARSER_TAG,
+  BodyParserFunction,
+} from './types';
+import {getContentType, normalizeParsingError} from './body-parser.helpers';
+import {RestBindings} from '../keys';
+import {JsonBodyParser} from './body-parser.json';
+import {UrlEncodedBodyParser} from './body-parser.urlencoded';
+import {TextBodyParser} from './body-parser.text';
+import {StreamBodyParser} from './body-parser.stream';
+import {RawBodyParser} from './body-parser.raw';
+
+export class RequestBodyParser {
+  private readonly parsers: BodyParser[];
+
+  constructor(
+    @inject(RestBindings.REQUEST_BODY_PARSER_OPTIONS, {optional: true})
+    options: RequestBodyParserOptions = {},
+    @inject.tag(REQUEST_BODY_PARSER_TAG, {optional: true})
+    parsers?: BodyParser[],
+    @inject.context() private readonly ctx?: Context,
+  ) {
+    if (!parsers || parsers.length === 0) {
+      this.parsers = [
```

Comment:
@raymondfeng let's discuss this part please before landing. If we land this list of default parsers, then it would require a semver-major version to remove it - i.e. it's a change that cannot be easily undone.